libs/langchain/langchain/document_loaders/recursive_url_loader.py

```diff
@@ -134,51 +129,34 @@ def _get_child_links_recursive(
             visited: A set of visited URLs.
         """
 
-        if depth > self.max_depth:
-            return []
-
-        # Add a trailing slash if not present
-        if not url.endswith("/"):
-            url += "/"
+        if depth >= self.max_depth:
+            return
+        # Exclude the links that start with any of the excluded directories
+        if any(url.startswith(exclude_dir) for exclude_dir in self.exclude_dirs):
+            return
 
         # Exclude the root and parent from a list
         visited = set() if visited is None else visited
 
-        # Exclude the links that start with any of the excluded directories
-        if self.exclude_dirs and any(
-            url.startswith(exclude_dir) for exclude_dir in self.exclude_dirs
-        ):
-            return []
-
         # Get all links that can be accessed from the current URL
         try:
             response = requests.get(url, timeout=self.timeout)
         except Exception:
-            return []
-
-        absolute_paths = self._get_sub_links(response.text, url)
+            logger.warning(f"Unable to load from {url}")
+            return
+        visited.add(url.rstrip("/"))
```

Comment:
technically a webserver may serve different content under `some_url` and `some_url/`